src/llmq/quorums_instantsend.cpp

```diff
@@ -1026,6 +1070,23 @@ void CInstantSendManager::TransactionAddedToMempool(const CTransactionRef& tx)
     }
 }
 
+void CInstantSendManager::TransactionRemovedFromMempool(const CTransactionRef& tx)
+{
+    if (!IsInstantSendEnabled() || !masternodeSync.IsBlockchainSynced() || tx->vin.empty()) {
```

Comment:
```suggestion
    if (tx->vin.empty()) {
```
Shouldn't it always at least try to remove it?